central/scannerdefinitions/handler/handler.go

```diff
@@ -654,3 +660,32 @@ func getOfflineFileVersion(mf *os.File) (string, error) {
 	}
 	return m.Version, nil
 }
+
+func validateV4DefsVersion(zipPath string) error {
+	zipR, err := zip.OpenReader(zipPath)
+	if err != nil {
+		return errors.Wrap(err, "couldn't open file as zip")
+	}
+	defer utils.IgnoreError(zipR.Close)
+
+	for _, zipF := range zipR.File {
+		if strings.HasPrefix(zipF.Name, scannerV4DefsPrefix) {
+			mf, err := openFromArchive(zipF.Name, "manifest.json")
+			if err != nil {
+				return errors.Wrap(err, "couldn't open v4 defs manifest.json")
+			}
+
+			offlineV, err := getOfflineFileVersion(mf)
+			if err != nil {
+				return errors.Wrap(err, "couldn't read v4 defs manifest.json")
+			}
+			utils.IgnoreError(mf.Close)
```

Comment:
Move this to be ahead of getOfflineFileVersion. L677

Reply:
maybe L678?
Will getOfflineFileVersion() have problem reading mf if mf has already been closed?

Reply:
That is for your deferred it. Can you close the last mf before returning in L680?